python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_run_launcher.py

```diff
@@ -83,10 +93,127 @@ def test_run_launcher_subset(self, graphql_context: WorkspaceRequestContext):
         assert result.data["pipelineRunOrError"]["status"] == "SUCCESS"
         assert result.data["pipelineRunOrError"]["stats"]["stepsSucceeded"] == 1
 
+    def test_run_launcher_unauthorized(self, graphql_context: WorkspaceRequestContext):
+        selector = infer_job_selector(graphql_context, "no_config_job")
 
-LaunchFailTestSuite: Any = make_graphql_context_test_suite(
-    context_variants=GraphQLContextVariant.all_non_launchable_variants()
-)
+        with patch.object(graphql_context, "has_permission_for_location", return_value=False):
+            with patch.object(graphql_context, "was_permission_checked", return_value=True):
+                result = execute_dagster_graphql(
+                    context=graphql_context,
+                    query=LAUNCH_PIPELINE_EXECUTION_MUTATION,
+                    variables={"executionParams": {"selector": selector, "mode": "default"}},
+                )
+                assert result.data["launchPipelineExecution"]["__typename"] == "UnauthorizedError"
+
+
+class TestMultipleLaunch(BaseTestSuite):
+    def test_multiple_run_launcher_same_job(self, graphql_context: WorkspaceRequestContext):
+        selector = infer_job_selector(graphql_context, "no_config_job")
+
+        # test with multiple of the same job
+        executionParamsList = [
+            {"selector": selector, "mode": "default"},
+            {"selector": selector, "mode": "default"},
+            {"selector": selector, "mode": "default"},
+        ]
+
+        result = execute_dagster_graphql(
+            context=graphql_context,
+            query=LAUNCH_MULTIPLE_RUNS_MUTATION,
+            variables={"executionParamsList": executionParamsList},
+        )
+
+        assert "launchMultipleRuns" in result.data
+        launches = result.data["launchMultipleRuns"]
+
+        assert launches["__typename"] == "LaunchMultipleRunsResult"
+        assert "launchMultipleRunsResult" in launches
+        results = launches["launchMultipleRunsResult"]
+
+        run_ids = []
+
+        for result in results:
+            assert result["__typename"] == "LaunchRunSuccess"
+            run_ids.append(result["run"]["runId"])
+
+        wait_for_runs_to_finish(graphql_context.instance)
+
+        for run_id in run_ids:
+            result = execute_dagster_graphql(
+                context=graphql_context, query=RUN_QUERY, variables={"runId": run_id}
+            )
+            assert result.data["pipelineRunOrError"]["__typename"] == "Run"
+            assert result.data["pipelineRunOrError"]["status"] == "SUCCESS"
```

Comment:
not sure we need this? I guess it's just asserting that the runs would have succeeded also if they were run individually but I think we can just assume that is the case given the mock setup.